// mailtrap/dist/lib/api/resources/Templates.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const config_1 = __importDefault(require("../../../config"));
const { CLIENT_SETTINGS } = config_1.default;
const { GENERAL_ENDPOINT } = CLIENT_SETTINGS;
class TemplatesApi {
    constructor(client, accountId) {
        this.client = client;
        this.templatesURL = `${GENERAL_ENDPOINT}/api/accounts/${accountId}/email_templates`;
    }
    /**
     * Get a list of all templates.
     */
    async getList() {
        const url = this.templatesURL;
        return this.client.get(url);
    }
    /**
     * Get a specific template by ID.
     */
    async get(templateId) {
        const url = `${this.templatesURL}/${templateId}`;
        return this.client.get(url);
    }
    /**
     * Create a new template.
     */
    async create(params) {
        const url = this.templatesURL;
        const data = { email_template: params };
        return this.client.post(url, data);
    }
    /**
     * Update an existing template.
     */
    async update(templateId, params) {
        const url = `${this.templatesURL}/${templateId}`;
        const data = { email_template: params };
        return this.client.patch(url, data);
    }
    /**
     * Delete a template.
     */
    async delete(templateId) {
        const url = `${this.templatesURL}/${templateId}`;
        return this.client.delete(url);
    }
}
exports.default = TemplatesApi;
